builtin/game/item_entity.lua

```diff
@@ -152,6 +178,111 @@ core.register_entity(":__builtin:item", {
 			return
 		end
 
+		local node_this = core.get_node_or_nil(pos)
+		local is_stuck = false
+		if node_this then
+			local sdef = minetest.registered_nodes[node_this.name]
+			is_stuck = (sdef.walkable == nil or sdef.walkable == true)
+				and (sdef.collision_box == nil or sdef.collision_box.type == "regular")
+				and (sdef.node_box == nil or sdef.node_box.type == "regular")
+		end
+
+		-- Push item out when stuck inside solid node
+		if is_stuck then
+			local shootdir
+			local cx = (pos.x % 1) - 0.5
+			local cz = (pos.z % 1) - 0.5
+			local order = {}
+
+			-- First prepare the order in which the 4 sides are to be checked.
```

Comment:
This code block that prepares an order seems overkill complex inside 'on step', why not just search the surrounding 4 or 8 nodes? and why not search diagonals?

Reply:
A fixed order might look pretty bad.
Imho, this code block doesn't look that complicated. Much more detailed calculations would be possible.

Reply:
What's bad about a fixed order? Maybe i've missed something ...

This is also about reducing code maintenance in future. This code seems like unnecessary headache.

Reply:
Well, with a fixed order, placing nodes into dropped items could be used as compass.

Such things are bad for player experience and the players will think that minetest code is simply bad and not well pollished.

Reply:
Clouds, node textures and probably several other things can be used as compasses too. And ... very few players think of MT as bad due to that.

When objects are buried, players are usually not aware of their position, so they don't need to choose the closest 'empty' node.